### PR TITLE
feat: add config for topRightNavigation

### PR DESCRIPTION
**Pull request by raintygao**
### Description

1. Add `opensearchDashboards.futureNavigation` config and control dev tool top right nav button.
2. Mark `rightNavigationButton` component as experimental.

## Screenshot

image
futureNavigation disabled by default

image
futureNavigation enabled

## Testing the changes
1. By default or add `opensearchDashboards.futureNavigation: false` in yml file, top right navigation will not display.
3. add `opensearchDashboards.futureNavigation: true` in yml file, top right navigation will display.

## Changelog

- feat: Add `opensearchDashboards.futureNavigation` config to control dev tool top right nav button.

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/console/server/plugin.ts [hunk 1]
```diff
@@ -66,10 +71,6 @@ export class ConsoleServerPlugin implements Plugin<ConsoleSetup, ConsoleStart> {
       });
     });
 
-    const config = await this.ctx.config.create().pipe(first()).toPromise();
```
Comment (ananzh) [inline src/plugins/console/server/plugin.ts:69]: Why we del this part?

Comment (raintygao) [reply src/plugins/console/server/plugin.ts:69]: Have been moved to top.

Comment (ananzh) [reply src/plugins/console/server/plugin.ts:69]: Oh I see. Thanks.

**Review by kavilla — APPROVED** [review]
have we looked into ui settings? might be a little easier.

File: changelogs/fragments/6712.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+feat:
+- Add config for topRightNavigation ([#6712](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/6712))
```
Comment (kavilla) [inline changelogs/fragments/6712.yml:2]: nit i would update the pr changelog to include the actual configuration and specify the dev tool top right nav button

File: src/core/server/opensearch_dashboards_config.ts [hunk 3]
```diff
@@ -91,6 +91,7 @@ export const config = {
         defaultValue: 'https://survey.opensearch.org',
       }),
     }),
+    topRightNavigation: schema.boolean({ defaultValue: false }),
```
Comment (kavilla) [inline src/core/server/opensearch_dashboards_config.ts:94]: nit: topRightNav* matches other references

**Review by kavilla — APPROVED** [review]

**Comment by raintygao** [comment]
> have we looked into ui settings? might be a little easier.

Thanks. That looks easier. But here feel like that we more need some configuration to be a role of feature flag. Like workspace feature flag.

File: changelogs/fragments/6712.yml (hunk 2 above)
Comment (raintygao) [reply changelogs/fragments/6712.yml]: Thank you, updated.

File: src/core/server/opensearch_dashboards_config.ts (hunk 3 above)
Comment (raintygao) [reply src/core/server/opensearch_dashboards_config.ts:94]: Should I update the name of configuration?  Do you have any recommendations?

Comment (raintygao) [reply src/core/server/opensearch_dashboards_config.ts:94]: Should I update the name of configuration? Do you have any recommendations for that?

**Review by SuZhou-Joe — APPROVED** [review]

**Comment by raintygao** [comment]
Seems  failing test is the same as main. https://github.com/opensearch-project/OpenSearch-Dashboards/actions/runs/8954435518/job/24594081770

**Review by ashwin-pc — CHANGES_REQUESTED** [review]
A few questions:
1. Why was this API introduced?
2. Cant the same feature flag that introduced this feature be used to turn this off?
3. Can we mark this API as experimental. The very existence of this PR suggests that its not clear if we even want to go with this approach.

**Review by SuZhou-Joe — APPROVED** [review]

**Review by ashwin-pc — APPROVED** [review]

**Comment by kavilla** [comment]
manual backport made here: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/6727

**Pull request merged**